### PR TITLE
Create log groups and log streams, log job output to cloudwatch logs

### PR DESCRIPTION
**Pull request by fishnix**
This creates log groups (per group) and log streams (per job) in Cloudwatch Logs.

Job output is logged to log streams.

### CODE REVIEW

File: cloudwatchlogs/cloudwatchlogs.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	if err != nil {
+		msg := fmt.Sprintf("failed to update retention policy for log group (%s)", group)
+		return ErrCode(msg, err)
+	}
+
+	return nil
+}
+
+// CreateLogStream creates a cloudwatchlogs log stream
+func (c *CloudWatchLogs) CreateLogStream(ctx context.Context, group, stream string) error {
```
Comment (damnski) [inline cloudwatchlogs/cloudwatchlogs.go]: group does not have a type

File: cloudwatchlogs/cloudwatchlogs.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+				return ErrCode(msg, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// LogEvent logs events to a log stream in a log group
+func (c *CloudWatchLogs) LogEvent(ctx context.Context, group, stream string, events []*Event) error {
```
Comment (damnski) [inline cloudwatchlogs/cloudwatchlogs.go]: similarly, group does not have a type defined

**Comment by damnski** [comment]
Wow - lot's to do to get logs.  Looks really good and massive unit tests.

File: cloudwatchlogs/cloudwatchlogs.go (hunk 1 above)
Comment (fishnix) [reply cloudwatchlogs/cloudwatchlogs.go]: it does - it's `string`.  you can omit the type for a list of variables of the same type

File: cloudwatchlogs/cloudwatchlogs.go (hunk 2 above)
Comment (fishnix) [reply cloudwatchlogs/cloudwatchlogs.go]: same same

File: cloudwatchlogs/cloudwatchlogs.go (hunk 1 above)
Comment (damnski) [reply cloudwatchlogs/cloudwatchlogs.go]: I see.

**Pull request merged**